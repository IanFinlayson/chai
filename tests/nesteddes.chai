
def printInfo(info [(String, Int)]):
    match info:
        case []:
            print("Done!")
        case ("Hammy", _) :: rest:
            print("If your name is Hammy, I don't care your age!")
            printInfo(rest)
        case (name, 0) :: rest:
            print(name, "is missing their age!")
            print("Giving up on recursion!")
        case (name, age) :: rest:
            print(name, "is", age, "years old.")
            printInfo(rest)
            
def main():
    printInfo([("Xavier", 26), ("Hammy", 13), ("Jean", 34), ("Bob", 0), ("Alice", 36)])

