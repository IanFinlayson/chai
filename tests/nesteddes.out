Xavier is 26 years old.
If your name is Hammy, I don't care your age!
Jean is 34 years old.
Bob is missing their age!
Giving up on recursion!
